pkg/util/discardedseries/tracker.go

```diff
@@ -0,0 +1,102 @@
+package discardedseries
+
+import (
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var trackedLabels = []string{
+	"sample_out_of_bounds",
+	"sample_out_of_order",
+	"sample_too_old",
+	"new_value_for_timestamp",
+	"per_user_series_limit",
+	"per_labelset_series_limit",
+	"per_metric_series_limit",
+}
```

Comment:
Can we avoid hardcoding the discarded reason. There can be more reasons in the future and it is very easy to miss it